File: src/main/java/pl/training/workshop/m6/s01_strategy/start/PriceBoard.java
package pl.training.workshop.m6.s01_strategy.start;

import pl.training.workshop.m6.s01_strategy.PriceRequest;
import pl.training.workshop.shared.Money;

/** Start: klient cennika - kasa wycenia bilet wg programu z konfiguracji. */
public final class PriceBoard {
    private final TicketPricer pricer = new TicketPricer();

    public Money priceFor(PriceRequest request) {
        return pricer.price(request.base(), request.ticketType(), request.program());
    }
}

File: csharp/src/Training.Workshop/M6/S01Strategy/Start/PriceBoard.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S01Strategy.Start;

/// <summary>Start: klient cennika - kasa wycenia bilet wg programu z konfiguracji.</summary>
public sealed class PriceBoard
{
    private readonly TicketPricer _pricer = new();

    public Money PriceFor(PriceRequest request)
    {
        return _pricer.Price(request.Base, request.TicketType, request.Program);
    }
}
